src/backend/functions/users.ts

```diff
@@ -97,7 +124,38 @@ const logUserIn: ApiRouteFunction = async (_params, event) => {
     }
   });
 
-  return buildSuccessResponse(usersTransformer(user));
+  return buildSuccessResponse(authenticatedUserTransformer(user));
+};
+
+/** Get settings for the specified user */
+const getUserSettings: ApiRouteFunction = async (params: { id: string }) => {
+  const userId: number = parseInt(params.id);
+  let user = await prisma.user.findUnique({
+    where: { userId },
+    include: { userSettings: true }
+  });
+  if (!user) return buildNotFoundResponse('user', `#${params.id}`);
+  if (!user.userSettings) {
+    user = await prisma.user.update({
+      where: { userId },
+      data: { userSettings: { create: {} } },
+      include: { userSettings: true }
+    });
+  }
+  return buildSuccessResponse(user.userSettings!);
+};
+
+/** Update settings for the specified user */
+const updateUserSettings: ApiRouteFunction = async (params: { id: string }, event) => {
```

Comment:
what if this user doesn't have settings yet? I guess we're relying on the user going to the settings page first which should call "getUserSettings" for them and create the settings object

Reply:
correct. I'm still maybe sorta a fan of just editing the data so there is no chance for a user to ever not have a user settings object. ignoring that though, the get request will create it. and the update form can't be accessed without calling the get request, so it's guaranteed to always exist for the update. the only scenario possible is if someone tries to update via API access. we could change to upsert. we would still need an answer for what to return if get has none though. lmao the more you know. halfway through writing this I googled it and: https://github.com/prisma/docs/issues/640

so basically either we
- edit data and assume it's always there
- leave it as is and say meh
- change both to upsert cuz that's most robust

Reply:
> change both to upsert cuz that's most robust

I picked this one cuz it's more robust and actually less code I think